### PR TITLE
feat(get-starknet): add skeleton to support get-starknet v4

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is to update the get-starknet wrapper to support the latest get-starknet v4 implementation, but in the meanwhile also supporting the legacy(v3) implementation

the Major change of the implementation are:

- having a new method `Init` -> to update the v3 property (account object, provider object, selectedAddress and isConnected) when network is switching
- `fix` a bug when the network switch but the address remain the same

### CODE REVIEW

File: packages/get-starknet/src/rpcs/switch-network.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { StarknetWalletRpc } from '../utils/rpc';
+
+export type WalletSwitchStarknetChainMethod = 'wallet_switchStarknetChain';
+type Params = RpcTypeToMessageMap[WalletSwitchStarknetChainMethod]['params'];
+type Result = RpcTypeToMessageMap[WalletSwitchStarknetChainMethod]['result'];
+
+export class WalletSwitchStarknetChain extends StarknetWalletRpc {
+  async handleRequest(param: Params): Promise<Result> {
+    try {
+      // The wallet.chainid should be always refer to the latest snap chain id
```
Comment (khanti42) [inline packages/get-starknet/src/rpcs/switch-network.ts:12]:
```suggestion
      // The wallet.chainid should always refer to the latest snap chain id
```

**Review by khanti42 — CHANGES_REQUESTED** [review]
Very minore changes, otherwise LGTM.

File: packages/get-starknet/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "prettier": "prettier --write \"src/**/*.ts\"",
     "lint": "eslint 'src/*.{js,ts,tsx}' --max-warnings 0 -f json -o eslint-report.json",
     "lint:fix": "eslint '**/*.{js,ts,tsx}' --fix",
-    "test:unit": ""
+    "test": "jest --passWithNoTests"
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply packages/get-starknet/package.json:21]: the github action is already using 'yarn test' from root, so it is fine

**Review by khanti42 — APPROVED** [review]
LGTM

**Pull request merged**